### PR TITLE
Update `fdpf2` stubs for v2.5.7

### LINKED ISSUE #8713: Stubtest failed on Fri Sep 09 2022
Stubtest runs are listed here: https://github.com/python/typeshed/actions/workflows/daily.yml

### PR DESCRIPTION
**Pull request by AlexWaygood**
Fixes #8713. A full diff between v2.5.6 and 2.5.7 can be seen here: https://github.com/PyFPDF/fpdf2/compare/2.5.6...2.5.7.

I tried to incorporate all changes to files currently included in typeshed's stubs, but decided not to try to add any new files in this PR.

### CODE REVIEW

File: stubs/fpdf2/fpdf/enums.pyi [hunk 1]
```diff
@@ -166,4 +178,10 @@ class Corner(CoerciveEnum):
     BOTTOM_RIGHT: str
     BOTTOM_LEFT: str
 
+class FontDescriptorFlag(Flag):
```
Comment (AlexWaygood) [inline stubs/fpdf2/fpdf/enums.pyi:181]:
```suggestion
class FontDescriptorFlags(Flag):
```

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 2]
```diff
@@ -47,9 +58,20 @@ class Annotation(NamedTuple):
...
 
+class EmbeddedFile(NamedTuple):
+    basename: str
+    bytes: bytes
+    desc: str = ...
+    creation_date: datetime.datetime | None = ...
+    modification_date: datetime.datetime | None = ...
+    compress: bool = ...
+    checksum: bool = ...
+    def file_spec(self, embedded_file_ref) -> dict[str, Incomplete]: ...
```
Comment (Akuli) [inline stubs/fpdf2/fpdf/fpdf.pyi:73]: `pdf_dict()` returns a string:

```suggestion
    def file_spec(self, embedded_file_ref) -> str: ...
```

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 3]
```diff
@@ -95,17 +117,21 @@ class FPDF:
...
     links: dict[int, DestinationXYZ]
+    embedded_files: list[Incomplete]
+    embedded_files_per_pdf_ref: dict[Incomplete, Incomplete]
     in_footer: int
     lasth: int
     current_font: _Font
     font_family: str
     font_style: str
+    font_stretching: float
+    char_spacing: int
```
Comment (Akuli) [inline stubs/fpdf2/fpdf/fpdf.pyi:128]: `set_char_spacing()` does `self._out(f"BT {spacing:.2f} Tc ET")` where `spacing` is a value of `self.char_spacing`, and the docstring says `float`.

```suggestion
    char_spacing: float
```

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 4]
```diff
@@ -250,13 +277,42 @@ class FPDF:
     ) -> None: ...
     def set_font(self, family: str | None = ..., style: _FontStyles = ..., size: int = ...) -> None: ...
     def set_font_size(self, size: float) -> None: ...
-    font_stretching: float
+    def set_char_spacing(self, spacing) -> None: ...
```
Comment (Akuli) [inline stubs/fpdf2/fpdf/fpdf.pyi:280]:
```suggestion
    def set_char_spacing(self, spacing: float) -> None: ...
```

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 5]
```diff
@@ -250,13 +277,42 @@ class FPDF:
...
     def set_link(self, link, y: int = ..., x: int = ..., page: int = ..., zoom: float | Literal["null"] = ...) -> None: ...
     def link(
         self, x: float, y: float, w: float, h: float, link: str | int, alt_text: str | None = ..., border_width: int = ...
     ) -> Annotation: ...
+    def embed_file(
+        self,
+        file_path: StrPath | None = ...,
+        bytes: bytes | None = ...,
+        basename: str | None = ...,
+        modification_date: datetime.datetime | None = ...,
```
Comment (Akuli) [inline stubs/fpdf2/fpdf/fpdf.pyi:292]: `creation_date` is missing. It is keyword-only and passed to `EmbeddedFile()`.

File: stubs/fpdf2/fpdf/fpdf.pyi [hunk 6]
```diff
@@ -250,13 +277,42 @@ class FPDF:
...
+        *,
+        desc: str = ...,
```
Comment (AlexWaygood) [inline stubs/fpdf2/fpdf/fpdf.pyi:294]:
```suggestion
        creation_date: datetime.datetime | None = ...,
        desc: str = ...,
```

**Review by Akuli — APPROVED** [review]

**Pull request merged**